engine/src/main/java/io/zeebe/engine/state/DefaultZeebeDbFactory.java

```diff
@@ -11,34 +11,68 @@
 import io.zeebe.db.ZeebeDbFactory;
 import io.zeebe.db.impl.rocksdb.ZeebeRocksDBMetricExporter;
 import io.zeebe.db.impl.rocksdb.ZeebeRocksDbFactory;
+import java.util.Properties;
 import java.util.function.BiFunction;
 
 public final class DefaultZeebeDbFactory {
 
+  public static final BiFunction<String, ZeebeDb<ZbColumnFamilies>, ZeebeRocksDBMetricExporter>
+      DEFAULT_DB_METRIC_EXPORTER_FACTORY =
+          (partitionId, database) ->
+              new ZeebeRocksDBMetricExporter<>(partitionId, database, ZbColumnFamilies.class);
+  private static ZeebeDbFactory<ZbColumnFamilies> instance;
+
   /**
-   * The default zeebe database factory, which is used in most of the places except for the
+   * Returns the default zeebe database factory, which is used in most of the places except for the
    * exporters.
+   *
+   * @return the created zeebe database factory
    */
-  public static final ZeebeDbFactory<ZbColumnFamilies> DEFAULT_DB_FACTORY =
-      defaultFactory(ZbColumnFamilies.class);
+  public static ZeebeDbFactory<ZbColumnFamilies> defaultFactory() {
+    return defaultFactory(new Properties());
+  }
 
-  public static final BiFunction<String, ZeebeDb<ZbColumnFamilies>, ZeebeRocksDBMetricExporter>
-      DEFAULT_DB_METRIC_EXPORTER_FACTORY =
-          (partitionId, database) -> {
-            return new ZeebeRocksDBMetricExporter<>(partitionId, database, ZbColumnFamilies.class);
-          };
+  /**
+   * Returns the default zeebe database factory, which is used in most of the places except for the
+   * exporters.
+   *
+   * @param userProvidedColumnFamilyOptions additional column family options
+   * @return the created zeebe database factory
+   */
+  public static synchronized ZeebeDbFactory<ZbColumnFamilies> defaultFactory(
+      final Properties userProvidedColumnFamilyOptions) {
+    if (instance == null) {
```

Comment:
You're right. I thought about it a bit before. The idea I had then was to make sure we wouldn't construct multiple factories, similar to the global constant. This might've made sense in case tests re-used the same factory, but I don't think any do. The method is not called twice, but when it is it shouldn't be a big deal. I'll just allow creating a new factory for every call.